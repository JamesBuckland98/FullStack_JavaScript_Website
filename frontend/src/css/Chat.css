li {
    padding: 2%;
    border-bottom: double;
    background-color: white;
}