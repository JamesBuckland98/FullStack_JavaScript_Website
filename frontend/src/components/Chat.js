import '../css/Chat.css'
import React, { Component } from 'react';
import { geolocated } from 'react-geolocated';

class Chat extends Component {

	componentDidMount() {
		this.scrollToBottom();
	}

	/**
	 * Method to scroll to the bottom message when component is mounted
	 * https://stackoverflow.com/questions/37620694/how-to-scroll-to-bottom-in-react
	 */
	scrollToBottom() {
		this.messagesEnd.scrollIntoView({ behavior: 'smooth' });
	}

	render() {
		return(
			<div>
				{
					this.props.messages.length > 0 ?
						<div>
							<ul> 
								{
									this.props.messages.map(message =>
										<li> 
											<i><b> { message.Username } : </b></i>
											<b> { message.ChatMessage } </b> <br/>
											<b> Species: { message.Species } </b> <br/>
											<b> Abundance: { message.Abundance} </b> <br/>
											<span> latitude : { message.Latitude }</span> 
											<span> longitude: { message.Longitude }</span>
											<span> Temp : { message.Weather }&deg;C </span>
											<i> 
											Date: { 
													new Date(message.ChatTimeStamp).toISOString().slice(0, 19).replace('T', ' ')
												} 
											</i>
										</li>
									)
								}
							</ul>
						</div>
						: null
				}
				<div 
					style={{ float:'left', clear: 'both' }}
					ref={(el) => { this.messagesEnd = el; }}
				/>
			</div>
		)
	}
}

export default geolocated({
	positionOptions: {
		enableHighAccuracy: false,
	},
	userDecisionTimeout: 5000,
})(Chat);