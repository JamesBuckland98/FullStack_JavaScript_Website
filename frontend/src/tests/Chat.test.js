import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Chat from '../components/Chat';
import { geolocated } from "react-geolocated";

it('should render chat component', () => {
	const mockSuccessfulGeolocationProvider = {
		getCurrentPosition(onSuccess) {
			return onSuccess({
				coords: {
					latitude: 50,
					longitude: 20,
				},
			});
		},
		watchPosition(onSuccess) {
			return onSuccess({
				coords: {
					latitude: 50,
					longitude: 20,
				},
			});
		},
	};
	const Component = geolocated({
		geolocationProvider: mockSuccessfulGeolocationProvider,
	})(Chat)
	const wrapper = shallow(<Component/>)
	expect(toJson(wrapper)).toMatchSnapshot();
})