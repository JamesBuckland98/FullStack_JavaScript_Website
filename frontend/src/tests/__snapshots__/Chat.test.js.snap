// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render chat component 1`] = `
<Geolocated(Chat)
  coords={
    Object {
      "latitude": 50,
      "longitude": 20,
    }
  }
  isGeolocationAvailable={true}
  isGeolocationEnabled={true}
  positionError={null}
/>
`;
